go/worker/keymanager/churp.go

```diff
@@ -130,7 +134,25 @@ func (w *churpWorker) Authorize(_ context.Context, method string, kind enclaverp
 	}
 
 	// All peers must undergo the authorization process.
-	return w.authorizeKeyManager(peerID)
+	switch method {
+	case churp.RPCMethodKeyShare:
+		return w.authorizeNode(ctx, peerID)
+	case churp.RPCMethodVerificationMatrix,
+		churp.RPCMethodShareReductionPoint,
+		churp.RPCMethodShareDistributionPoint,
+		churp.RPCMethodBivariateShare:
+		return w.authorizeKeyManager(peerID)
+	default:
+		return fmt.Errorf("unsupported method: %s", method)
+	}
+}
+
+func (w *churpWorker) authorizeNode(_ context.Context, peerID core.PeerID) error {
+	// TODO: Which nodes are allowed to query key shares?
```

Comment:
I refactored domain separation used by CHURP. The context/dst used to encode a `key_id` is now:
- `oasis-core/keymanager/churp: encode SGX policy key ID for runtime <kmID> for churp <ID>` for the first bullet, and
- `oasis-core/keymanager/churp: encode custom policy key ID for runtime <kmID> for churp <ID>` for the second one.

This will encode/hash `key_id` to a different group element, thus creating different shares. Chain context is not taken into account, but maybe it should be? Since churp ID uniquely defines the group and the hash function, we don't need to include that in the domain separation. 

Access policy (executor + observer) is checked on the host node, just like it's implemented for the `secrets`. Key manager runtime only checks that remote enclave is allowed to query key shares for the given runtime ID.

Reply:
> Chain context is not taken into account, but maybe it should be?

What would happen during a possible dump/restore upgrade, would this invalidate all of the keys? We need to be careful.

> Access policy (executor + observer) is checked on the host node, just like it's implemented for the secrets. Key manager runtime only checks that remote enclave is allowed to query key shares for the given runtime ID.

Yeah checks for roles inside the runtime can be added in the future when everyone is using observer nodes.

Reply:
> What would happen during a possible dump/restore upgrade, would this invalidate all of the keys? We need to be careful.

Yes, the keys would of course change. Will keep as it is.

> Yeah checks for roles inside the runtime can be added in the future when everyone is using observer nodes.

Ok, we can add this latter.